docs/developer/contributing.md

```diff
@@ -0,0 +1,83 @@
+
+#### Reporting Bugs
+  If you encounter a bug, please search for existing issues on the issue tracker first.
+
+  If you can't find a duplicate issue, open a new one.
+
+  Follow the template and provide a clear description of the bug, including steps to reproduce it if possible.
+  Screenshots, logs, or code snippets can also be helpful.
+
+#### Suggesting Features
+  Do you have an idea on how Pumpkin can be improved? Share your thoughts by opening an issue on the issue tracker.
+
+  Describe the proposed feature in detail, including its benefits and potential implementation considerations.
+
+#### Contributing Code
+  To get started with contributing code to Pumpkin, fork the repository on github
+
+- First create a github account if you don't already have one
+ 
+- Go to Pumpkin's Official [Github Repository](https://github.com/Snowiiii/Pumpkin) and press fork
+
+> Creating a fork means you now have your own copy of the Pumpkin source code, (this does not mean you own the copyright)
+
+  Now that you have a copy that you can edit, you will need a few tools
+
+- Install [Github-Desktop](https://desktop.github.com/download/) or [git](https://git-scm.com/downloads) for your operating system
+
+> Github Desktop is a much easier tool to use. If you are unsure of which to use, choose this.
+
+- Install Rust at [rust-lang.org](https://www.rust-lang.org/).
+
+- Finally, if you are wanting to contribute to documentation, install [NodeJS](https://nodejs.org/en)
+
+#### Github Desktop
+  If you chose to use Github Desktop and don't know what a token is, press enter your password and follow the steps on screen
+
+- Enter the password you chose for Github
+
+- A few options will be made avaliable, select **Clone Repository**
+
+- Click on the fork that you created, it should say something like **{YourUsername}/Pumpkin**
+
+- Select where you want your fork to be placed on your computer
+
+- Click **Clone**
+
+  Whenever you make changes to Pumpkin's code, they will appear within Github Desktop. Click **Commit to Master** to submit them to your fork
+
+  On Github you can view your fork, if it says you are behind in contributions, click **Sync Fork** and **Update Branch**
+
+  When you are ready to submit your code and make it official, press **Contribute**, then **Open Pull Request**
+
+#### Git
+  If you chose to use git, clone your repository and start contributing
+
+- Use **git clone https://github.com/{YourUsername}/Pumpkin** to clone the repository
+
+- Use **git add --all** or **git add {fileChanged}** to add files to the repository that weren't there previously
+
+- Use **git commit** to Commit your code to Master
+
+  On Github you can view your fork, if it says you are behind in contributions, click **Sync Fork** and **Update Branch**
+
+- Use **git pull** to update your local code
+
+  Ensure your code adheres to our project structure and style guidelines.
+
+  Write clear and concise commit messages that describe your changes.
```

Comment:
I think its is way too much, There are many explanations for how to use git/github desktop out there